Examples/Algorithms/TrackFinding/src/TrackParamsEstimationAlgorithm.cpp

```diff
@@ -64,60 +73,130 @@ Acts::Vector3 ActsExamples::TrackParamsEstimationAlgorithm::getField(
       m_cfg.magneticField);
 }
 
+ActsExamples::SimSeedContainer
+ActsExamples::TrackParamsEstimationAlgorithm::createSeeds(
+    const ActsExamples::ProtoTrackContainer& protoTracks,
+    const ActsExamples::SimSpacePointContainer& spacePoints) const {
+  SimSeedContainer seeds;
+  seeds.reserve(protoTracks.size());
+  for (std::size_t itrack = 0; itrack < protoTracks.size(); ++itrack) {
+    // The list of hits and the initial start parameters
+    const auto& protoTrack = protoTracks[itrack];
+    if (protoTrack.size() < 3) {
+      ACTS_WARNING("Proto track " << itrack << " size is less than 3.");
+      continue;
+    }
+    // Space points on the proto track
+    std::vector<SimSpacePoint> spacePointsOnTrack;
+    spacePointsOnTrack.reserve(protoTrack.size());
+    // Loop over the hit index on the proto track to find the space points
+    for (const auto& hitIndex : protoTrack) {
+      auto it =
+          std::find_if(spacePoints.begin(), spacePoints.end(),
+                       [&](const SimSpacePoint& spacePoint) {
+                         return (spacePoint.measurementIndex() == hitIndex);
+                       });
+      if (it != spacePoints.end()) {
+        spacePointsOnTrack.push_back(*it);
+      }
+    }
+    // At least three space points are required
+    if (spacePointsOnTrack.size() < 3) {
+      continue;
+    }
+    // Sort the space points
+    std::sort(spacePointsOnTrack.begin(), spacePointsOnTrack.end(),
+              [](const SimSpacePoint& lhs, const SimSpacePoint& rhs) {
+                return lhs.r() < rhs.r();
```

Comment:
Here and in the following lines you use r for the ordering but wouldn't the absolute distance be better (r^2+z^2) ? I could see case in the forward region where due to resolution effect, r appear to decrease when you get further away.